pkg/p2p/libp2p/internal/handshake/handshake.go

```diff
@@ -128,13 +134,16 @@ func (s *Service) Handshake(stream p2p.Stream, peerMultiaddr ma.Multiaddr, peerI
 			Overlay:   bzzAddress.Overlay.Bytes(),
 			Signature: bzzAddress.Signature,
 		},
-		NetworkID: s.networkID,
-		Light:     s.lightNode,
+		NetworkID:      s.networkID,
+		Light:          s.lightNode,
+		WelcomeMessage: s.welcomeMessage,
 	}); err != nil {
 		return nil, fmt.Errorf("write ack message: %w", err)
 	}
 
 	s.logger.Tracef("handshake finished for peer %s", remoteBzzAddress.Overlay.String())
+	s.logger.Tracef("received welcome message %s from peer", resp.Ack.WelcomeMessage)
```

Comment:
This should be logged only if welcome message is not blank. And maybe to log the peer overlay address at the end. Also Info log level suggested for better visibility of the message as it is the default log level.